packages/cli/src/spec/mock/MockPromptForStrings.ts

```diff
@@ -0,0 +1,107 @@
+   * @param message the message to be prompt. Not relevant for tests.
+   * @param {string | null} defaultValue a default value or null.
+   * @param {ValidateFunction<T>} validateFunction a function to validate the input.
+   * @returns {Promise<T>} a {@link Promise} that resolves to the validated loaded message,
+   * converted to `T` by the `validateFunction`.
+   */
+  async promptInput<T>(_message: string,
+      _defaultValue: string | null,
+      validateFunction: ValidateFunction<T>): Promise<T> {
+    const nextResponse = this.getNextMessage();
+    return (await validateFunction(nextResponse)).unwrap();
+  }
+
+  /**
+   * Sets the output to be the given message.
+   * @param message the message to be prompt. Not relevant for tests.
+   * @param {string[]} choices a list of choices. Not relevant for testing.
+   * @param {string | null} defaultValue a default value or null.
+   * @param {ValidateFunction<T>} validateFunction a function to validate the input.
+   * @returns {Promise<T>} a {@link Promise} that resolves to the validated loaded message,
+   * converted to `T` by the `validateFunction`.
+   */
+  async promptChoice<T>(_message: string,
+      _choices: string[],
+      _defaultValue: string | null,
+      validateFunction: ValidateFunction<T>): Promise<T> {
+    const nextResponse = this.getNextMessage();
+    return (await validateFunction(nextResponse)).unwrap();
+  }
+
+  /**
+   * Sets the output to be the given message.
+   * @param message the message to be prompt. Not relevant for tests.
+   * @param defaultValue the value to be returned
+   * @returns {Promise<boolean>} a {@link Promise} that resolves to a {@link boolean} value. The
+   * value will the `true` if the user answers `Yes` and `false` for `No`.
+   */
+  async promptConfirm(_message: string, defaultValue: boolean): Promise<boolean> {
+    return defaultValue;
+  }
+
+  /**
+   * Sets the output to be the given message.
+   * @param message the message to be prompt. Not relevant for tests.
+   * @param {ValidateFunction<T>} validateFunction a function to validate the input.
+   * @returns {Promise<string>} a {@link Promise} that resolves to the user input validated by
+   * `validateFunction`.
+   */
+  async promptPassword(_message: string, validateFunction: ValidateFunction<string>,
+  ): Promise<string> {
+    const nextResponse = this.getNextMessage();
+    return (await validateFunction(nextResponse)).unwrap();
+  }
+
+  /**
+   * Sets the output to be the given message.
+   * @param {ValidateFunction<T>} validateFunction a function to validate the input.
+   * @returns {string} which is the next message to be prompted`.
+   */
+  private getNextMessage(): string {
+    if (this.responses.length < 0) {
+      throw new Error('No answer was given. Please use addMessage(nextMessage) before' +
+      ' using this function');
+    }
+    const nextResponse = this.responses[this.responses.length];
+    this.responses.shift();
```

Comment:
I still don't think the logic is correct here, you're adding to the front of the list, removing from the front of the list and reading from the back of the list.

```
addMessage("hello") // responses = ["hello"]
addMessage("world") // responses = ["world", "hello"]
getMessage() // returns "hello", responses = ["hello"]
getMessage() // returns "hello", responses = []
```

Could you write a test? Just a simple Spec that adds a few things and calls one of the prompts to make sure the logic and ordering is correct.

(Actually come to think of it, won't this always crash? If you've got an array of length `1`, you're going to try to access `responses[1]`, which is out of bounds.)

Reply:
You are right, I will change the logic

Reply:
Should I upload the test or do it just for me?

Reply:
Upload the test, it would be nice to have a proper Spec for it.